common.go

```diff
@@ -154,3 +155,18 @@ func secureRandomBytes(bytes uint) ([]byte, error) {
 	_, err := rand.Read(r)
 	return r, err
 }
+
+// secureCompare will compare two slice of bytes in constant time, ensuring no timing information
+// is leaked in order to prevent timing attacks.
+func secureCompare(a, b []byte) bool {
+	if len(a) != len(b) {
```

Comment:
Can we rely on [crypto/subtle's ConstantTimeCompare](https://golang.org/pkg/crypto/subtle/#ConstantTimeCompare)?

Should this be used for `User.CheckPassword` in integration_test.go? Otherwise I don't see anywhere it's used and we could remove altogether.